src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/impl/GitClientImpl.java

```diff
@@ -618,101 +697,141 @@ public boolean checkoutFeatureBranch(Ref featureBranch) {
+    private Ref getRef(String jiraIssueKey, String repoUri) {
+	List<Ref> refs = getAllRefsForOneRepo(repoUri);
 	Ref branch = null;
 	for (Ref ref : refs) {
 	    if (ref.getName().contains(jiraIssueKey)) {
 		return ref;
-	    } else if (ref.getName().equalsIgnoreCase("refs/heads/develop")) {
-		branch = ref;
-	    } else if (ref.getName().equalsIgnoreCase("refs/heads/master")) {
+	    } else if (ref.getName().equalsIgnoreCase("refs/heads/" + defaultBranchFolderNames.get(repoUri))) {
 		branch = ref;
 	    }
 	}
 	return branch;
     }
 
-    private List<Ref> getAllRefs() {
-	return getRefs(ListBranchCommand.ListMode.ALL);
+    @Override
+    public List<Ref> getAllRemoteBranches() {
+	return getAllRefs(ListBranchCommand.ListMode.REMOTE);
     }
 
     @Override
-    public List<Ref> getRemoteBranches() {
-	return getRefs(ListBranchCommand.ListMode.REMOTE);
+    public List<Ref> getRemoteBranches(String repoUri) {
+	return getRefs(ListBranchCommand.ListMode.REMOTE, repoUri);
     }
 
-    private List<Ref> getRefs(ListBranchCommand.ListMode listMode) {
+    private List<Ref> getAllRefsForOneRepo(String repoUri) {
+	return getRefs(ListBranchCommand.ListMode.ALL, repoUri);
+    }
+
+    private List<Ref> getRefs(ListBranchCommand.ListMode listMode, String repoUri) {
 	List<Ref> refs = new ArrayList<Ref>();
 	try {
-	    refs = git.branchList().setListMode(listMode).call();
+	    refs = gits.get(repoUri).branchList().setListMode(listMode).call();
 	} catch (GitAPIException | NullPointerException e) {
 	    LOGGER.error("Git could not get references. Message: " + e.getMessage());
 	}
 	return refs;
     }
 
-    private List<RevCommit> getCommitsFromDefaultBranch() {
-	return getCommits(defaultBranch, true);
+    private List<Ref> getAllRefs(ListBranchCommand.ListMode listMode) {
+	List<Ref> allRefs = new ArrayList<Ref>();
+	for (String uri : remoteUris) {
+	    try {
+		List<Ref> refs = gits.get(uri).branchList().setListMode(listMode).call();
+		allRefs.addAll(refs);
+	    } catch (GitAPIException | NullPointerException e) {
+		LOGGER.error("Git could not get references. Message: " + e.getMessage());
+	    }
+	}
+
+	return allRefs;
+    }
+
+    public List<RevCommit> getAllRelatedCommits(Issue jiraIssue) {
+	List<RevCommit> allCommits = new ArrayList<RevCommit>();
+	for (String repoUri : remoteUris) {
+	    List<RevCommit> com = getCommits(jiraIssue, repoUri);
+	    allCommits.addAll(com);
+	}
+	return allCommits;
+    }
+
+    private List<RevCommit> getCommitsFromDefaultBranch(String repoUri) {
+	return getCommits(repoUri, defaultBranches.get(repoUri), true);
     }
 
     private List<RevCommit> getCommits(Ref branch) {
-	return getCommits(branch, false);
+	String repoUri = getRepoUriFromBranch(branch);
+	return getCommits(repoUri, branch, false);
     }
 
-    private List<RevCommit> getCommits(Ref branch, boolean isDefaultBranch) {
+    private List<RevCommit> getCommits(String repoUri, Ref branch, boolean isDefaultBranch) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'getCommits(String, Ref, boolean)' has an NPath complexity of 528](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=5160185)